### ISSUE TITLE
How to launch processes to the specified logical cpus on eatch node

### ISSUE THREAD

**Issue by XiaotaoChen** [open-mpi/ompi#6299]
I'm using openmpi 4.0 to launch my distributed program. I want to bind the specified cores to the processes launched by mpi.  So i can assign the remaining cores to other threads(such as communication thread). i try to bind the specified cores with `-rf rankfile`. ref this docs: https://www.open-mpi.org/doc/current/man1/mpirun.1.php  but it tells error:
```
ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612
ORTE_ERROR_LOG: Data unpack would read past end of buffer in file base/odls_base_default_fns.c at line 612
```

### my hardware
```
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                80
On-line CPU(s) list:   0-79
Thread(s) per core:    2
Core(s) per socket:    20
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz
Stepping:              4
CPU MHz:               2379.199
CPU max MHz:           3700.0000
CPU min MHz:           1000.0000
BogoMIPS:              4800.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              28160K
NUMA node0 CPU(s):     0-19,40-59
NUMA node1 CPU(s):     20-39,60-79
```
I want to launch processes on each socket. And each porcess on use 18 physical cores on the socket. So i set  `slot=0-17` . And the remaining 2 cores can bind to communication thread.
#### my launch scripts as below:
```
/home/wuxunzha/softwares/openmpi4.0/bin/mpirun \
    -np 2 \
    -H 192.168.20.126 \
    -rf ${myrankfile} \
    --report-bindings \
    -x MXNET_USE_OPERATOR_TUNING=0 \
    -x HOROVOD_FUSION_THRESHOLD=0 \
    -mca pml ob1 \
    -mca btl ^openib \
    -mca btl_tcp_if_include 192.168.20.1/24 \
    -mca rmaps_rank_file_physical 1 \
    ${workspace}/${single_script}
```
#### rankfile info:
```
rank 0=192.168.20.126 slot=0:0-17
rank 1=192.168.20.126 slot=1:0-17
```

**Comment by thananon** [comment]
This looks legit to me and it should work. It has been a couple days since you posted. I'm not sure if you found the solution yet.

I would try this workaround.
` mpirun -np 2 --map-by socket --bind-to socket ./y our_app`. While this will bind each process to the whole socket, if you have less thread than the number of physical core, you should get similar performance.

**Comment by XiaotaoChen** [comment]
@thananon thanks for your suggestion. `bind-to socket` means each process use the whole cores on one socket. in my case, it will use 20 cores instead of 18 cores. And this option `bind-to socket` can work normally.

**Comment by rhc54** [comment]
This was previously reported by another user and resolved in the upcoming v4.0.1 release - see #6203 for the PR that went into master and #6204 for the one into the v4.0.x branch
